diff --git a/setuptools/package_index.py b/setuptools/package_index.py
index 9c9d76a..b641d21 100644
--- a/setuptools/package_index.py
+++ b/setuptools/package_index.py
@@ -232,7 +232,10 @@ class HashChecker(ContentChecker):
 
     def __init__(self, hash_name, expected):
         self.hash_name = hash_name
-        self.hash = hashlib.new(hash_name)
+        if hash_name == 'md5':
+            self.hash = hashlib.new(hash_name, usedforsecurity=False)
+        else:
+            self.hash = hashlib.new(hash_name)
         self.expected = expected
 
     @classmethod
